plugins/woocommerce/tests/api-core-tests/tests/settings/settings-crud.test.js

```diff
@@ -239,7 +239,7 @@ test.describe('Settings API tests: CRUD', () => {
 
 	test.describe('List all settings options', () => {
 
-		test.fixme('can retrieve all general settings', async ({
```

Comment:
I wonder if we may want to leave this as is, or at least update some of the tests. I am seeing some errors when running this against a LocalWP instance, for example:

<img width="1001" alt="Screen Shot 2022-11-09 at 11 00 36" src="https://user-images.githubusercontent.com/71906536/200906896-89633348-051e-4c52-a7a7-144681013b85.png">

<img width="1085" alt="Screen Shot 2022-11-09 at 11 00 15" src="https://user-images.githubusercontent.com/71906536/200906899-b13dc7c3-ddc1-4895-8376-879a76b15554.png">

So far it seems to work as expected in the JN and local environments, but there's still some potential flakiness here. Primarily I am seeing it around the `can retrieve all advanced settings` test. Perhaps we could skip these for now until we come up with a good fix, or update the tests on this PR to be less strict on the response.

Let me know your thoughts!

Here's the setup I have just for example if you wanted to try and reproduce this behavior:
* Installed [LocalWP](https://localwp.com)
* Set up a test site with the defaults
* Installed WooCommerce
* Installed basic auth plugin
* Updated the local `.env` file to use the credentials for the site I set up
* Ran the tests

Seems to be happening fairly consistency after the first time I ran through the tests against the site. The first run all passed, but subsequent runs have all passed on the same step above.

Reply:
Curious on your thoughts for an approach here @nigeljamesstevenson. And, hopefully, it's something reproducible!

Reply:
Thanks a lot for looking into this @zhongruige - especially for checking out the execution with the [LocalWP](https://localwp.com/) - that wasn't something I had tried before and a great tool to assist.

I downloaded and installed LocalWP and created a site choosing the Preferred environment and defaults, installed WooCommerce and basic auth plugin and updated the .env.

The tests flew along at a nice pace thanks to the local env and failed on the following:

- `can retrieve all products settings` and 
- `can retrieve all advanced settings`and 
- `can view all system status items`

So - good that it is reproducible 😄 
....and we definitely need to make these tests more solid.

 There are differences between ids on shop pages so I will make them less strict.
 e.g. Localhost woocommerce_cart_page_id:
```
       "value": "6",
        "options": {
            "2": "Sample Page",
            "5": "Shop",
            "6": "Cart",
            "7": "Checkout",
            "8": "My account"
        },
```
LocalWP woocommerce_cart_page_id::
```
       "value": "7",
        "options": {
            "2": "Sample Page",
            "6": "Shop",
            "7": "Cart",
            "8": "Checkout",
            "9": "My account"
        },
```
 
I would be interested to know if there were matchers for the object keys, for example:
**KEYMATCHER**: "Shop",
**KEYMATCHER**: "Cart", etc
 but after some attempts - and a spot of googling am not 100%. For now, I will make the options sections match with any object...

Reply:
Note - the failure on view all system status items was due to the fact there were no inactive plugins. I do not feel this test is valid and am therefore going to remove it (not all sites will have inactive plugins)

Reply:
Thanks for the LocalWP pointer!!!

Reply:
Oh - gross! No - I didn't see a fail on this one - let me look into it!

Reply:
As a matter of interest, was it local, JN or localWP it failed against?

Reply:
I noticed it only against LocalWP, but I didn't run it against JN yet to see if it's happening there, too.

Reply:
I wonder if it's because of using the same site (not a fresh install), since I am getting failures on JN now that passed before (but different compared to LocalWP). So perhaps this is a bit of a red herring.

Reply:
This may be a red herring and if the (previously more flaky) tests failed and caused the env to be off kilter with slightly dodgy data then that makes sense but regardless - the fresh v non-fresh install is a good call out. Let me take a look see! We want these to be solid across localhost/JN/LocalWP and also repeatable. 
Thanks again for the input!